ADR/ADR037-architecture-of-review-apps.md

```diff
@@ -0,0 +1,120 @@
+
+### What is in use elsewhere in GDS?
+
+Other teams in GDS have faced the same issues as us. One particularly good example is the GOV.UK Design System team, who have a mix of technical and non-technical members. They host prototypes and review versions of applications on [Heroku](https://www.heroku.com/) using Heroku's [Review Apps](https://devcenter.heroku.com/articles/github-integration-review-apps) feature. This is also in on GOV.UK Forms among our content and interaction designers for hosting prototypes.
+
+We investigated using Heroku Review Apps for our purposes, but we concluded that it was not a good choice. Heroku Review Apps is aimed at use cases involving a single application, but the applications we want to get review apps for will require deploying more than one application at a time.
+
+## Decision
+
+In this section we describe the architectural decisions we've made. Implementation details are out of scope.
+
+### Workflow 
+
+We have decided that the workflow for working with review apps should look like this
+
+```mermaid
+%%{init: { "sequence": { "showSequenceNumbers": true }}}%%
+        
+sequenceDiagram
+
+actor dev as Dev
+actor reviewer as Reviwer
+participant pr as PR
+participant app as Review App
+
+dev ->> pr: open
+activate pr
+
+pr ->> app: create
+activate app
+
+pr -->> dev: review app URL
+
+loop Iteration
+    dev ->> pr: new commit
+    pr ->> app: redeploy
+    dev ->> app: visit
+end
+
+reviewer ->> pr: review
+pr -->> reviewer: review app URL
+reviewer ->> app: visit
+reviewer ->> pr: approve
+
+dev ->> pr: merge
+pr ->> app: destroy
+deactivate app
+
+pr -->> dev: merged
+deactivate pr
+```
+
+A developer (or non-technical team member) will raise a pull request for their change, and the system we build will deploy a review app in response to the opening of the pull request. The system will share a link to the running app in a comment on the pull request.
+
+Subsequent changes to the pull request will result in the app being redeployed.
+
+Once a reviewer has reviewed and merged the pull request, the system will destroy the running applications. 
+
+### Architecture 
+Given the workflow we want to have, and the investigations we have done, we have decided that we should build a review apps solution ourselves, using common services available to us in AWS. Below is high level architecture diagram for we what we have settled on.
+![A diagram showing the proposed architecture for GOV.UK Forms review apps](../diagrams/images/ADR037-architecture-diagram.png)
+
+We will run the review apps themselves as containers on AWS ECS on Fargate. The task definition we use will deploy containerised PostgreSQL alongside `forms-addmin` and `forms-api`, and configure the apps to talk to the local database. This will ensure that the database in use for review apps is not shared between different instances of the review apps. [^seeding] 
```

Comment:
Minor: Typo:
```suggestion
We will run the review apps themselves as containers on AWS ECS on Fargate. The task definition we use will deploy containerised PostgreSQL alongside `forms-admin` and `forms-api`, and configure the apps to talk to the local database. This will ensure that the database in use for review apps is not shared between different instances of the review apps. [^seeding] 
```